Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 33340)
+++ CMakeLists.txt	(working copy)
@@ -802,7 +802,7 @@
 endif()
 
 casa_find( CASACORE
-           VERSION 2.0.2
+           VERSION 2.0.0
            PREFIX_HINTS ${CASACORE_ROOT_DIR} ${Boost_INCLUDE_DIR}
            INCLUDES casa/aips.h
                     casa/version.h
@@ -1074,14 +1074,14 @@
   #set( LIBSAKURA_ROOT_DIR "$ENV{HOME}/workspace/libsakura/build/installed" )
   set( LIBSAKURA_ROOT_DIR ${casa_packages} )
 endif()
-casa_find( LIBSAKURA
-  VERSION 0.1
-  INCLUDES libsakura/sakura.h libsakura/config.h
-  LIBS sakura
-  PREFIX_HINTS ${LIBSAKURA_ROOT_DIR} ${LIBSAKURA_ROOT_DIR}/lib/libsakura/default
-  CPP_VERSION "LIBSAKURA_VERSION_MAJOR << '.' << LIBSAKURA_VERSION_MINOR"
-  NO_LINK    # The library may depend on newer gcc. Skip generating executable.
-)
+#casa_find( LIBSAKURA
+#  VERSION 0.1
+#  INCLUDES libsakura/sakura.h libsakura/config.h
+#  LIBS sakura
+#  PREFIX_HINTS ${LIBSAKURA_ROOT_DIR} ${LIBSAKURA_ROOT_DIR}/lib/libsakura/default
+#  CPP_VERSION "LIBSAKURA_VERSION_MAJOR << '.' << LIBSAKURA_VERSION_MINOR"
+#  NO_LINK    # The library may depend on newer gcc. Skip generating executable.
+#)
 
 casa_config_end()
 
@@ -1122,8 +1122,8 @@
 casa_add_module( dish CASACORE Boost)
 casa_add_module( nrao CASACORE Boost)
 casa_add_module( miriad CASACORE Boost components)
-casa_add_module( casa_sakura CASACORE LIBSAKURA)
-casa_add_module( singledish CASACORE casa_sakura msvis mstransform)
+#casa_add_module( casa_sakura CASACORE LIBSAKURA)
+#casa_add_module( singledish CASACORE casa_sakura msvis mstransform)
 casa_add_module( bnmin1 Boost GSL GSLCBLAS CASACORE )
 casa_add_module( air_casawvr CASACORE bnmin1 Boost synthesis WCSLIB CFITSIO LAPACK )
 
